test/jdk/javax/swing/JTableHeader/SerializeJTableHeader.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 6450193
+ * @summary JTableHeader doesn't uninstall UI after first serialization
+ * @run main SerializeJTableHeader
+ */
+
+import javax.swing.table.JTableHeader;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+import javax.swing.SwingUtilities;
+
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectOutputStream;
+
+public class SerializeJTableHeader {
+
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+        } catch (UnsupportedLookAndFeelException ignored) {
+            System.out.println("Unsupported L&F: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException
+                 | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
+            System.out.println("Testing L&F: " + laf);
+            SwingUtilities.invokeAndWait(() -> setLookAndFeel(laf));
```

Comment:
Why did you choose to set LAF on EDT? You then continue with creating and serialising the component on the main thread.

Reply:
I was not sure if any event processing needs to be done for lookandfeel change particularly native l&f like GTK and Aqua.
Seems like it works with both EDT and non-EDT..tried with repeat count 50..
however GTK is not supported on headless environment so made the test run on headful environment.

Reply:
I never thought about it… Yet if there's anything that depends on access to native platform, it probably needs to run on the toolkit thread.

I'm for keeping it simpler: if it works correctly on the main thread here, let's keep it this way.